### ISSUE TITLE
Add "Run Windows" label to all PRs that impact Windows related components

### ISSUE THREAD

**Issue by crobert-1** [open-telemetry/opentelemetry-collector-contrib#27866]
### Describe the issue you're reporting

We currently require the `Run Windows` label to be added to PRs to run tests on Windows. This label can only be added by triagers, approvers, or maintainers. It would be good to automate the `Run Windows` label being added to relevant PRs.

This will reduce the chance of the label being missed on PRs when relevant, as well getting rid of the frequent step of contributors asking these other members to add the label.

**Comment by mx-psi** [comment]
Should we have a field on `metadata.yaml` specifying this? Maybe a list of supported OS?

**Comment by crobert-1** [comment]
A list of supported OSs in `metadata.yaml` sounds like a good idea going forward. It may be hard to initially setup with so many components, but it would hold up to changes the best over time. I'm sure it would be helpful for end users as well to know what's supported. Down the road version support could be added as well but that's probably a discussion for another day.

This will end up running Windows tests more often than _entirely_ required, but I assume that is acceptable.

**Comment by crobert-1** [comment]
I was able to get some more context for this issue since I filed it. It sounds like the `Run Windows` label was created originally as a workaround for flaky tests when running on Windows. There's no real concern about running tests on Windows, other than flakiness. The end goal should be to fix test flakiness when running on Windows and always run tests on Windows. 

It may still be valid to include some kind of supported OSes in `metadata.yaml`, but not so much for this purpose.
